projects/packages/publicize/src/jetpack-social-settings/class-settings.php

```diff
@@ -46,15 +46,26 @@ class Settings {
 	 * @return void
 	 */
 	private function migrate_old_option() {
-		$auto_conversion_settings = get_option( 'jetpack_social_settings' );
-		if ( ! empty( $auto_conversion_settings ) ) {
-			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => ! empty( $auto_conversion_settings['image'] ) ) );
+		// Migrating from the old option.
+		$old_auto_conversion_settings = get_option( 'jetpack_social_settings' );
+		if ( ! empty( $old_auto_conversion_settings ) ) {
+			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, array( 'enabled' => ! empty( $old_auto_conversion_settings['image'] ) ) );
 			delete_option( 'jetpack_social_settings' );
 		}
+		// Checking if the new option is valid.
+		$auto_conversion_settings = get_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES );
+		if ( ! is_array( $auto_conversion_settings ) || ! isset( $auto_conversion_settings['enabled'] ) ) {
+			update_option( self::OPTION_PREFIX . self::AUTOCONVERT_IMAGES, self::DEFAULT_AUTOCONVERT_IMAGES_SETTINGS );
```

Comment:
Should we actually delete it instead? We'll start using the defaults everywhere and it stops anyone that has the option set from this bug having it stored in their database.

Reply:
Good idea, I'll move to that